internal/app/secretless/providers/kubernetessecrets/client.go

```diff
@@ -4,47 +4,33 @@ import (
 	"k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/kubernetes"
-	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/rest"
 )
 
-// KubeClient represents Kubernetes client and calculated namespace
+// KubeClient represents a Kubernetes client
 type KubeClient struct {
```

Comment:
does this still need to be a struct? why not `type KubeClient kubernetes.Interface`

Reply:
@izgeri the struct is a wrapper with the convenience method `#getSecret`. e.g. it'd make it easier to add any logic to check if the environment is Openshift, or whatever else, without exposing all that in the provider. I suppose it could also just be a free method called `#getSecret` that takes `kc kubernetes.Interface` and `secretName string`

Reply:
Agreed that would be cleaner. Alas, interfaces are a different type of animal and can't have methods at all^ :(